### PR TITLE
Add player animations

### PR DESCRIPTION
**Pull request by abosiljcic**
Player can run and stand. When the game starts the player is standing by default. When the left/right arrows are clicked, the player runs. These features are added to the struct MovementControlSystem in Main.cpp.

### CODE REVIEW

File: src/Main.cpp [hunk 1]
```diff
@@ -5,170 +5,126 @@
...
-};
-
-struct Player {};
-
-struct Ball 
+struct Movement 
 {
-	geometry::Vec2 speed;
+	geometry::Vec2 velocity;
+	geometry::Vec2 velocitymax;
```
Comment (gavrilovmiroslav) [inline src/Main.cpp:11]: `velocity_max` and `move_force` plez

File: src/Main.cpp [hunk 2]
```diff
@@ -5,170 +5,126 @@
...
-	KeyCode down;
+	KeyCode left, down, up, right;
 };
 
-struct PhysicsSystem
-	: public ecs::System
-	, public AccessGroupStorage<Player, Position>
-	, public AccessGroupStorage<Ball, Position>
-	, public SignalEmitter<PadCollisionSignal>
+static float fsignf(float x)
```
Comment (gavrilovmiroslav) [inline src/Main.cpp:23]: Could we put all of these functions into a header? Also, why are these all `static`?

File: src/Main.cpp [hunk 3]
```diff
@@ -5,170 +5,126 @@
...
 		const auto& keys = KeyState::get();
 
-		if (keys.is_pressed(KEY_ESCAPE))
+		if (keys.is_down(KeyCode::KEY_ESCAPE))
 		{
 			Engine::get_instance().quit();
 		}
 
-		for (auto&& [entity, bindings, pos] : access_storage().each())
+		for (auto&& [entity, keybinding, movement, sprite] : access_storage().each())
```
Comment (gavrilovmiroslav) [inline src/Main.cpp:76]: I'd also like `key_binding`; whenever you have multiple words, basically, cut it up

**Review by gavrilovmiroslav — CHANGES_REQUESTED** [review]

File: src/Main.cpp [hunk 4]
```diff
@@ -5,170 +5,126 @@
...
 	: public ecs::System
-	, public MutAccessGroupStorage<Player, KeyBindings, Position>
+	, public MutAccessGroupStorage<KeyBindings, Movement, SpriteAnimation>
 {
 	void on_tick() override
 	{
 		const auto& keys = KeyState::get();
 
-		if (keys.is_pressed(KEY_ESCAPE))
+		if (keys.is_down(KeyCode::KEY_ESCAPE))
```
Comment (Mentathiel) [inline src/Main.cpp:71]: Exiting the game doesn't belong with movement controls, it should be in a system together with Pause or other game states that controls the overall game flow.

File: src/Main.cpp [hunk 5]
```diff
@@ -5,170 +5,126 @@
...
 		{
 			Engine::get_instance().quit();
 		}
 
-		for (auto&& [entity, bindings, pos] : access_storage().each())
+		for (auto&& [entity, keybinding, movement, sprite] : access_storage().each())
 		{
-			if (keys.is_down(bindings.up))
+			if (keys.is_down(keybinding.up))         
+				movement.force.y = -movement.moveforce;
```
Comment (Mentathiel) [inline src/Main.cpp:79]: Add curly braces even on "one liners" everywhere in code, it's easier to modify later and less error-prone.

**Review by Mentathiel — CHANGES_REQUESTED** [review]

**Review by milica-vucic — APPROVED** [review]

**Comment by gavrilovmiroslav** [comment]
@harry-cpp this build fail is somewhat weird, don't you think?

**Review by Mentathiel — APPROVED** [review]

**Comment by harry-cpp** [comment]
> @harry-cpp this build fail is somewhat weird, don't you think?

The fork just needs to be resynced with main.

**Review by gavrilovmiroslav — APPROVED** [review]
Mostly merge changes, next time merge directly no need to wait

**Pull request merged**